File: src/main/java/EarningsEntry.java
/**
 * Data for one upcoming (or recently-past) earnings event.
 *
 * <p>Fields that are unknown or not yet reported are stored as
 * {@link Double#NaN} so callers can display "N/A" without special-casing nulls.
 *
 * @param ticker        upper-case ticker symbol, e.g. "AAPL"
 * @param companyName   long company name, or ticker if unavailable
 * @param earningsDate  Unix epoch seconds of the expected earnings date; 0 if unknown
 * @param earningsTime  "BMO" (before open), "AMC" (after close), or "—" if unknown
 * @param epsEstimate   analyst consensus EPS estimate; {@link Double#NaN} if unavailable
 * @param epsActual     reported EPS; {@link Double#NaN} if not yet announced
 */
public record EarningsEntry(
        String ticker,
        String companyName,
        long   earningsDate,
        String earningsTime,
        double epsEstimate,
        double epsActual
) {}

File: MobilePortStock-iOS/MobilePortStock/Models/EarningsEntry.swift
import Foundation

/// Data for one upcoming (or recently-past) earnings event.
struct EarningsEntry: Codable, Identifiable, Hashable {
    /// Upper-case ticker symbol, e.g. "AAPL".
    let ticker: String

    /// Long company name, or ticker if unavailable.
    let companyName: String

    /// Unix epoch seconds of the expected earnings date; `nil` if unknown.
    let earningsDate: TimeInterval?

    /// "BMO" (before open), "AMC" (after close), or "—" if unknown.
    let earningsTime: String

    /// Analyst consensus EPS estimate; `nil` if unavailable.
    let epsEstimate: Double?

    /// Reported EPS; `nil` if not yet announced.
    let epsActual: Double?

    var id: String { ticker }

    /// Earnings date as a `Date` for display formatting.
    var earningsDateFormatted: Date? {
        guard let earningsDate else { return nil }
        return Date(timeIntervalSince1970: earningsDate)
    }
}
